src/main/java/ch/njol/skript/test/platform/PlatformMain.java

```diff
@@ -134,19 +140,24 @@ public static void main(String... args) throws IOException, InterruptedException
 		
 		// All succeeded tests in a single line
 		System.out.printf("%s Results %s%n", StringUtils.repeat("-", 25), StringUtils.repeat("-", 25));
-		System.out.println("Tested environments: " + String.join(", ",
+		System.out.println("\nTested environments: " + String.join(", ",
 				envs.stream().map(Environment::getName).collect(Collectors.toList())));
-		System.out.println("\nSucceeded: " + String.join(", ", succeeded));
+		System.out.println("\nSucceeded:\n  " + String.join((junit ? "\n  " : ", "), succeeded));
+
 		if (!failNames.isEmpty()) { // More space for failed tests, they're important
-			System.err.println("Failed:");
+			Thread.sleep(1); // speed at which system prints fails without this, and can look like successful tests failed.
+			System.err.println("\nFailed:");
 			for (String failed : failNames) {
 				List<NonNullPair<Environment, String>> errors = failures.get(failed);
-				System.err.println("  " + failed + " (on " + errors.size() + " environments)");
+				System.err.println("  " + failed + " (on " + errors.size() + " environment" + (errors.size() == 1 ? "" : "s") + ")");
 				for (NonNullPair<Environment, String> error : errors) {
 					System.err.println("    " + error.getSecond() + " (on " + error.getFirst().getName() + ")");
 				}
 			}
-			System.exit(failNames.size()); // Error code to indicate how many tests failed
+			Thread.sleep(1);
```

Comment:
Flush the stream instead of some magical sleep.

Reply:
I think you misunderstand, the `Thread.sleep(1)` is bad and should be replaced with flushing the output stream.

Reply:
The reasoning is because when the console writes, the messages get mixed up with eachother due to race condition. A string builder would be a better solution.